### ISSUE TITLE
Fresh installation of pixeltable in a docker container fails due to missing directories 

### ISSUE THREAD

**Issue by zhenzhongxu** [pixeltable/pixeltable#412]
Repro environment:
- Docker (codespace, debian bookworm)
- conda create -n pxt python=3.12
- pip install pixeltable
- pxt.init() or pxt.drop_dir() throws below error:

```
PermissionError                           Traceback (most recent call last)
Cell In[4], [line 4](vscode-notebook-cell:?execution_count=4&line=4)
      [1](vscode-notebook-cell:?execution_count=4&line=1) import pixeltable as pxt
      [3](vscode-notebook-cell:?execution_count=4&line=3) # Create the directory `demo` (if it doesn't already exist)
----> [4](vscode-notebook-cell:?execution_count=4&line=4) pxt.drop_dir('demo1', force=True)  # First drop `demo` to ensure a clean environment
      [5](vscode-notebook-cell:?execution_count=4&line=5) # pxt.create_dir('demo')
      [6](vscode-notebook-cell:?execution_count=4&line=6) 
      [7](vscode-notebook-cell:?execution_count=4&line=7) # # Create the table `demo.first` with a single column `input_image`
      [8](vscode-notebook-cell:?execution_count=4&line=8) # t = pxt.create_table('demo.first', {'input_image': pxt.Image})

File /opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:442, in drop_dir(path_str, force, ignore_errors)
    [422](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:422) def drop_dir(path_str: str, force: bool = False, ignore_errors: bool = False) -> None:
    [423](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:423)     """Remove a directory.
    [424](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:424) 
    [425](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:425)     Args:
   (...)
    [440](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:440)         >>> pxt.drop_dir('my_dir.sub_dir')
    [441](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:441)     """
--> [442](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:442)     cat = Catalog.get()
    [443](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:443)     path = catalog.Path(path_str)
    [445](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/globals.py:445)     try:

File /opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:28, in Catalog.get(cls)
     [25](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:25) @classmethod
     [26](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:26) def get(cls) -> Catalog:
     [27](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:27)     if cls._instance is None:
---> [28](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:28)         cls._instance = cls()
     [29](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:29)         with orm.Session(env.Env.get().engine, future=True) as session:
     [30](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:30)             cls._instance._load_table_versions(session)

File /opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:48, in Catalog.__init__(self)
     [45](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:45) self.tbls: dict[UUID, Table] = {}  # don't use a defaultdict here, it doesn't cooperate with the debugger
     [46](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:46) self.tbl_dependents: dict[UUID, list[Table]] = {}
---> [48](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:48) self._init_store()
     [49](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:49) self.paths = PathDict()

File /opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:53, in Catalog._init_store(self)
     [51](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:51) def _init_store(self) -> None:
     [52](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:52)     """One-time initialization of the stored catalog. Idempotent."""
---> [53](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:53)     with orm.Session(env.Env.get().engine, future=True) as session:
     [54](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:54)         if session.query(sql.func.count(schema.Dir.id)).scalar() > 0:
     [55](https://vscode-remote+codespaces-002bopulent-002dspace-002dcarnival-002dpgw7x7gwgr39j5.vscode-resource.vscode-cdn.net/opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/catalog/catalog.py:55)             return

File /opt/conda/envs/pai-dev/lib/python3.12/site-packages/pixeltable/env.py:75, in Env.get(cls)
...
File <frozen os>:215, in makedirs(name, mode, exist_ok)

File <frozen os>:225, in makedirs(name, mode, exist_ok)

PermissionError: [Errno 13] Permission denied: b'/run/user'
```

Running the below commands manually fixed the issue, but it was not a great initial user experience as the issue seem to be buried  [here](https://github.com/pixeltable/pixeltable/blob/main/.github/workflows/nightly.yml#L70). I suggest either properly document it (for now), or fix it properly as many dev environments may vary.

```
sudo mkdir -p /run/user/1000 /run/user/1001
sudo chmod a+rwx /run/user/1000 /run/user/1001
```

**Comment by aaron-siegel** [comment]
Thanks for reporting this. We had thought this was an issue specific to the Github Actions runners, but evidently not. We'll look into it.

Aaron

**Comment by aaron-siegel** [comment]
This is fixed in 0.2.29 (just released). Please reopen if you have any further issues. Thanks again for the bug report!

Aaron
